contracts/strategy/liquity/LiquityStrategy.sol

```diff
@@ -211,86 +213,71 @@ contract LiquityStrategy is
     }
 
     /**
-     * Collects the LQTY & ETH rewards from the stability pool, swaps the rewards to LUSD,
-     * and reinvests swapped LUSD amount into the stability pool to create compound interest on future gains.
-     *
-     * @notice Can only be called by the account with the ADMIN role.
-     * @notice Implicitly calls the reinvestRewards function.
-     * @notice Arguments provided to harvest function are real-time data obtained from '0x' api.
-     *
-     * @param _swapTarget the address of the '0x' contract performing the tokens swap.
-     * @param _lqtySwapData data used to perform LQTY -> LUSD swap.
-     * @param _ethSwapData data used to perform ETH -> LUSD swap.
+     * Collects the LQTY & ETH rewards from the stability pool.
      */
-    function harvest(
-        address _swapTarget,
-        bytes calldata _lqtySwapData,
-        bytes calldata _ethSwapData
-    ) external virtual {
-        // call withdrawFromSP with 0 amount only to claim rewards
+    function harvest() external virtual {
+        // call to withdrawFromSP with 0 amount will only claim rewards
         stabilityPool.withdrawFromSP(0);
-
-        reinvestRewards(_swapTarget, _lqtySwapData, _ethSwapData);
     }
 
     /**
-     * Swaps LQTY tokens and ETH already held by the strategy to LUSD,
-     * and reinvests swapped LUSD amount into the stability pool.
+     * Swaps LQTY tokens and ETH held by the strategy to LUSD,
+     * and reinvests the swapped LUSD amount into the stability pool.
      *
-     * @notice Can only be called by the account with the ADMIN role.
-     * @notice Arguments provided are real-time data obtained from '0x' api.
+     * @notice Can only be called by the account with the KEEPER role.
+     * @notice Swap data arguments provided are real-time data obtained from '0x' api.
      *
      * @param _swapTarget the address of the '0x' contract performing the tokens swap.
-     * @param _lqtySwapData data used to perform LQTY -> LUSD swap.
-     * @param _ethSwapData data used to perform ETH -> LUSD swap.
+     * @param _lqtySwapData data used to perform LQTY -> LUSD swap. Leave empty to skip this swap.
+     * @param _ethAmount amount of ETH to swap to LUSD, has to match with the amount used to obtain @param _ethSwapData.
+     * @param _ethSwapData data used to perform ETH -> LUSD swap. Leave empty to skip this swap.
      */
-    function reinvestRewards(
+    function reinvest(
         address _swapTarget,
+        uint256 _lqtyAmount,
         bytes calldata _lqtySwapData,
+        uint256 _ethAmount,
         bytes calldata _ethSwapData
     ) public virtual onlyKeeper {
```

Comment:
I guess this can be external now?

Reply:
it can be public actually :slightly_smiling_face: